pdns/ueberbackend.cc

```diff
@@ -368,8 +377,8 @@ bool UeberBackend::getAuth(const DNSName &target, const QType& qtype, SOAData* s
       if(i == backends.end()) {
         if(d_negcache_ttl) {
           DLOG(g_log<<Logger::Error<<"add neg cache entry:"<<shorter<<endl);
-          d_question.qname=shorter;
-          addNegCache(d_question);
+          d_question.qname = shorter;
+          addNegCache(d_question, d_question.qtype);
```

Comment:
I'm confused, how would we know the requested qtype then, so that we can filter the records in `UeberBackend::get()`, if we don't store it in `d_question.qtype`? `d_handle.qtype` is already set to `ANY` in that case so we do the correct lookup when we iterate over backends in `UeberBackend::handle::get()`.